--- urdfdom/urdf_parser/CMakeLists.txt.orig	2022-05-28 18:26:52.112844417 +0200
+++ urdfdom/urdf_parser/CMakeLists.txt	2022-05-28 22:28:05.201701811 +0200
@@ -18,7 +18,9 @@
   if(NOT CMAKE_CXX_STANDARD)
     target_compile_features(${add_urdfdom_library_LIBNAME} PUBLIC cxx_std_14)
   endif()
+  target_compile_definitions(${add_urdfdom_library_LIBNAME} PUBLIC "TIXML_USE_STL")
   set_target_properties(${add_urdfdom_library_LIBNAME} PROPERTIES
+    INSTALL_RPATH "\$ORIGIN"
     DEFINE_SYMBOL URDFDOM_EXPORTS
     SOVERSION ${URDF_MAJOR_MINOR_VERSION})
 endmacro()
